kpi/permissions.py

```diff
@@ -416,14 +416,19 @@ def has_object_permission(self, request, view, obj):
 
 
 class EditSubmissionPermission(EditLinkSubmissionPermission):
-
+    # TODO: Refactor this so we don't have to check for the object twice
     def has_permission(self, request, view):
         try:
             return super().has_permission(request, view)
-        except Http404:
-            # When we receive a 404, we want to force a 401 to let the user
-            # log in with different credentials. Enketo Express will prompt
-            # the credential form only if it receives a 401.
+        except Http404 as e:
+            uid = request.parser_context['kwargs']['uid']
+            # Is this a real 404 (object does not exist)? If so, raise it
+            if not AssetSnapshot.objects.filter(uid=uid).exists():
+                raise e
```

Comment:
As we observed in the past, we should not add `e` with we want to re-raise an exception without modifying it. 

See See https://chat.kobotoolbox.org/#narrow/stream/4-Kobo-Dev/topic/Python.20coding.20style/near/331618 for more details.